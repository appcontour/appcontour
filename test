#!/bin/bash
#

l__ist=`ls tests/test.??`
if [ -n "$1" ]
then
  l__ist=tests/test.$1
fi

tmpf__ile=test$$.tmp
failed="0"

for f__ile in $l__ist
do
  i__nfo=$(grep "^#testname:" $f__ile | cut -f2- -d:)
  if [ -n "$i__nfo" ]; then i__nfo=" ($i__nfo)"; fi
  test=`echo $f__ile | cut -f2 -d\.`
  echo "Running test $test$i__nfo..."
  source $f__ile 2>/dev/null >$tmpf__ile
  exitcode=$?

  if [ -f ${f__ile}.out ]
  then
    if ! diff $tmpf__ile ${f__ile}.out
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying out file"
    cp $tmpf__ile ${f__ile}.out
  fi
  rm $tmpf__ile
  echo "$exitcode" >$tmpf__ile
  if [ -f ${f__ile}.ec ]
  then
    if ! diff $tmpf__ile ${f__ile}.ec
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying ec file"
    cp $tmpf__ile ${f__ile}.ec
  fi
  rm $tmpf__ile
done

if [ "$failed" -gt "0" ]
then
  echo $failed tests failed
  exit 1
fi

exit 0
