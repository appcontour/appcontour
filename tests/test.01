./contour applyrule n1 <examples/tre_sfere.morse
