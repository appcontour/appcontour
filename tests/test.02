#./contour applyrule n1 <examples/three_spheres.morse | ./contour applyrule n1 | ./contour applyrule n1 | ./contour applyrule n1
./contour applyrule k2 <examples/three_spheres.morse | ./contour applyrule k2 | ./contour applyrule k2 | ./contour applyrule k2
