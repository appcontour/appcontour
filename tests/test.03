./contour applyrule n1 <examples/tre_sfere.morse | ./contour applyrule n1 | ./contour applyrule n1
