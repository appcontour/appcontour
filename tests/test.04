./contour ishuffman -q examples/two_rings.morse
./contour iscontour -q examples/immersed/boysurface.morse
