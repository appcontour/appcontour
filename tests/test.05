./contour applyrule n2 <examples/sfera_dietro_toro.morse
./contour applyrule n3 <examples/sfera_davanti_toro.morse
./contour applyrule n4 <examples/due_anelli.morse | ./contour applyrule n2 | ./contour applyrule n3 | ./contour applyrule n1
./contour applyrule n4 <examples/toro2.morse
./contour applyrule N5 <examples/tre_sfere.morse | ./contour applyrule N5:6 | ./contour applyrule N1 | ./contour applyrule N1 | ./contour applyrule N1
