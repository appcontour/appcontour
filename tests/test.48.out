Handlebody links up to 6 crossings
Handlebody links with two components: HL4_1 HL5_1 HL6_1 HL6_2 HL6_3 HL6_4 HL6_5 HL6_6 HL6_7 HL6_8
SPLIT link:A4:178
4_1:A4:114
5_1:A4:98
6_1:A4:90
6_2:A4:106
6_3:A4:90
6_6:A4:130
6_7:A4:98
6_8:A4:114
SPLIT link:A5:3675
4_1:A5:600
5_1:A5:660
6_1:A5:600
6_3:A5:469
6_7:A5:597
6_8:A5:1401
SPLIT link with HK4_1:A4:274
6_4:A4:106
6_5:A4:210
Handlebody links with three components: HL6_9 HL6_10 HL6_11 HL6_12 HL6_13 HL6_14
6_9:A4:310
6_10:A4:326
6_11:A4:486
6_11_fake:A4:694
6_12:A4:502
6_13:A4:822
6_14:A4:486
6_15:A4:1242
Handlebody links with four components: HL6_15
